### DISCUSSION TITLE
deno.worker

### DISCUSSION THREAD

**Discussion by unicomp21** [nats-io/nats.deno#274] [Q&A] [answered]
can I run nats.deno in a browser worker, under deno.worker ?

https://deno.land/manual/typescript/configuration

**Comment by aricart** [comment]
Haven't tried it but if you enable the Deno namespaces to work on the worker, my guess is yes.

**Comment by unicomp21** [comment]
I don't know the current state of things on the deno side, my current focus is on using esfx w/ webpack.

**Comment by aricart** [comment]
@unicomp21 and @gedw99 nats.deno doesn't run in a browser. But nats.ws does! - the API is exactly the same. As for running in a worker, I have not tried running nats.ws in a worker, but Deno most certainly does.

https://github.com/aricart/services_demo/blob/main/service-adm.ts#L204

**Reply by aricart** [reply]
Not sure why you would... We have nats.ws for exactly that reason..

**Reply by aricart** [reply]
With that said the question is whether you are doing multiple connections or processing in the workers.

**Reply by aricart** [reply, marked as answer]
You can call `connect()` from a worker and effectively that is an independent connection parallelizing within the same process. In my apps I have used multiple workers each with its own connection.
